### DISCUSSION TITLE
Does keda-operator ClusterRole need access to serviceaccounts?

### DISCUSSION THREAD

**Discussion by gdamjan-loka** [kedacore/keda#2348] [Q&A / Need Help] [answered]
Does the the `keda-operator` `ClusterRole` need access to `serviceaccounts` resources too?

I was getting these errors:
```
E1125 13:48:04.476470       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.8/tools/cache/reflector.go:167: Failed to watch *v1.ServiceAccount: failed to list *v1.ServiceAccount: serviceaccounts is forbidden: User "system:serviceaccount:keda:keda-operator" cannot list resource "serviceaccounts" in API group "" at the cluster scope
```

To get rid of the error I had to add `serviceaccount` here somewhere (I edited the release yaml file):
https://github.com/kedacore/keda/blob/main/config/rbac/role.yaml#L19-L23

**Comment by tomkerkhove** [comment]
What version of KEDA are you running?

**Reply by gdamjan-loka** [reply]
> What version of KEDA are you running?

2.4.0

**Reply by tomkerkhove** [reply]
Would you mind giving 2.5 a quick try please?

However, this looks like a bug given it was removed in our latest Helm resource update ([link](https://github.com/kedacore/charts/pull/205/files?authenticity_token=L8Zko8twT1kVfYFGL0%2B9rplq%2FoUPVOAu8sUpxzs%2FeuiBW0EIiKwQRaYOCtf3pHu2y9G7ScT1rocm7WVhdQcnVg%3D%3D&file-filters%5B%5D=.md&file-filters%5B%5D=.yaml#), which was not fully aligned for v2.4 as well) - Thoughts @zroubalik ?

**Reply by dominhhai** [reply]
v2.5.0 has same error here
```
E1204 14:51:09.838845       1 reflector.go:138] pkg/mod/k8s.io/client-go@v0.22.4/tools/cache/reflector.go:167: Failed to watch *v1.ServiceAccount: failed to list *v1.ServiceAccount: serviceaccounts is forbidden: User "system:serviceaccount:keda:keda-operator" cannot list resource "serviceaccounts" in API group "" at the cluster scope
```

How can I pass this error?

**Reply by eidmantas** [reply]
@tomkerkhove  With 2.5.0 same error here as well. Can not list serviceaccounts. What should be the correct action here? Is it really necessary?

**Reply by tomkerkhove** [reply]
@zroubalik I think we need it, no?

**Comment by gdamjan-loka** [comment, marked as answer]
Didn't have the chance to test this, but I believe it's fixed with this PR https://github.com/kedacore/keda/pull/2406
